go/kbfs/libkbfs/conflict_resolver.go

```diff
@@ -3183,6 +3194,121 @@ outer:
 	return nil
 }
 
+const conflictRecordVersion = 1
+
+type conflictRecord struct {
+	Version int
+	time.Time
+	Merged      string
+	Unmerged    string
+	ErrorTime   time.Time
+	ErrorString string
+	PanicString string
+	codec.UnknownFieldSetHandler
+}
+
+func (cr *ConflictResolver) recordStartResolve(ci conflictInput) error {
+	db := cr.config.KBFSOps().GetConflictResolutionDB()
+	conflictsSoFarSerialized, err := db.Get([]byte(cr.fbo.id().String()), nil)
+	var conflictsSoFar []conflictRecord
+	switch err {
+	case leveldb.ErrNotFound:
+		conflictsSoFar = nil
+	case nil:
+		err = cr.config.Codec().Decode(conflictsSoFarSerialized,
+			&conflictsSoFar)
+		if err != nil {
+			return err
+		}
+	default:
+		return err
+	}
+	if len(conflictsSoFar) > 10 {
+		conflictsSoFar = conflictsSoFar[len(conflictsSoFar)-10:]
+	}
+	conflictsSoFar = append(conflictsSoFar, conflictRecord{
+		Version:  conflictRecordVersion,
+		Time:     cr.config.Clock().Now(),
+		Merged:   ci.merged.String(),
+		Unmerged: ci.unmerged.String(),
+	})
+	conflictsSerialized, err := cr.config.Codec().Encode(conflictsSoFar)
+	if err != nil {
+		return err
+	}
+	return db.Put([]byte(cr.fbo.id().String()), conflictsSerialized, nil)
+}
+
+// recordFinishResolve does one of two things:
+//  - in the event of success, it deletes the DB entry that recorded conflict
+//    resolution attempts for this resolver
+//  - in the event of failure, it logs that CR failed and tries to record the
+//    failure to the DB.
+func (cr *ConflictResolver) recordFinishResolve(
+	ctx context.Context, ci conflictInput, receivedErr error) {
+	db := cr.config.KBFSOps().GetConflictResolutionDB()
+	panicVar := recover()
+
+	// If we neither errored nor panicked, this CR succeeded and we can wipe
+	// the DB entry.
+	if receivedErr == nil && panicVar == nil {
```

Comment:
@jakob223 Oh one more thing: I think this should be something like:

```go
if (receivedErr == nil || receivedErr == context.Canceled) && panicVar == nil {
```

Because legitimate cancellations happen a lot, in fact they happen every time a new write is done to the unmerged branch.  So they should basically be treated like nil errors.  (You can also just modify the caller of this function to pass in a `nil` err in that case, if you think that's cleaner.)